signer/services.go

```diff
@@ -1,17 +1,85 @@
 package signer
 
 import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
 	"sync"
+	"syscall"
 
 	tmLog "github.com/tendermint/tendermint/libs/log"
 	tmOS "github.com/tendermint/tendermint/libs/os"
 	tmService "github.com/tendermint/tendermint/libs/service"
 )
 
-func WaitAndTerminate(logger tmLog.Logger, services []tmService.Service) {
+func RequireNotRunning(pidFilePath string) error {
+	if _, err := os.Stat(pidFilePath); err != nil {
+		if os.IsNotExist(err) {
+			// lock file does not exist, can continue starting daemon
+			// or performing other tasks that require horcrux daemon to be stopped.
+			return nil
+		}
+		return fmt.Errorf("unexpected error while checking for existence of lock file at %s: %w", pidFilePath, err)
+	}
+
+	lockFile, err := os.ReadFile(pidFilePath)
+	if err != nil {
+		return fmt.Errorf("error reading lock file: %s, %w", pidFilePath, err)
+	}
+
+	pid, err := strconv.ParseInt(strings.TrimSpace(string(lockFile)), 10, 64)
+	if err != nil {
+		return fmt.Errorf("unexpected error parsing PID from lock file: %s. manual deletion of PID file required. %w",
+			pidFilePath, err)
+	}
+
```

Comment:
Should we compare `pid` with `os.Getpid()` here? I'm not sure what action that would take -- maybe even panicking would be okay.

Reply:
Added the check. Panic makes sense since it indicates a future coding error, so the stack trace will be helpful there.